apps/web/src/hooks/useForm.ts

```diff
@@ -48,10 +48,23 @@ export const useForm = () => {
     const todoForms: FormInstanceEntity[] = assignedFIData.filter(
       (formInstance: FormInstanceEntity) => {
         const signatures: SignatureEntity[] = formInstance.signatures;
-        for (let i = 0; i < signatures.length; i++) {
-          if (signatures[i].signerPositionId === user.positionId) {
-            return signatures[i].signed === false;
-          }
+
+        // Filter out signatures that are already signed
+        signatures.filter((signature: SignatureEntity) => {
+          return signature.signed === false;
+        });
+
+        // Sort signatures by order
+        signatures.sort((a: SignatureEntity, b: SignatureEntity) => {
```

Comment:
i think a better order of logic would be:
1. sort by order
2. go through the sorted list of signatures, and find the first one that isn't signed
3. return true if the first found corresponds to the user's positionId, else false